src/app/src/advancedsetupdialog.h

```diff
@@ -0,0 +1,22 @@
+#ifndef  ADVANCEDSETUPDIALOG_H
+#define  ADVANCEDSETUPDIALOG_H
+
+#include <QDialog>
+#include <QListWidget>
+#include <QListWidgetItem>
+
+namespace Ui {
+class AdvancedSetupDialog;
+}
+
+class AdvancedSetupDialog : public QDialog{
+    Q_OBJECT
+
+public:
+    explicit AdvancedSetupDialog(QWidget *parent = 0);
+    ~AdvancedSetupDialog();
+
+private:
+    Ui::AdvancedSetupDialog *ui;
```

Comment:
probably no reason to make this a pointer since it's only used in this class and has to get constructed when the class is initialized anyway

Reply:
oh, I see, you're setting it up the same way I did the stagecontrol, I don't remember why I did it that way but you probably don't need to do it that way